src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+        public void TestArgumentExceptionsAsyncEnumerable()
+        {
+            Assert.Throws<ArgumentNullException>(() => new TransformManyBlock<int, int>((Func<int, IAsyncEnumerable<int>>)null));
+            Assert.Throws<ArgumentNullException>(() => new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, null));
+
+            DataflowTestHelpers.TestArgumentsExceptions(new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable));
+        }
+
+        [Fact]
+        public void TestToStringAsyncEnumerable()
+        {
+            DataflowTestHelpers.TestToString(nameFormat =>
+                nameFormat != null ?
+                    new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, new ExecutionDataflowBlockOptions() { NameFormat = nameFormat }) :
+                    new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable));
+        }
+
+        [Fact]
+        public async Task TestOfferMessageAsyncEnumerable()
+        {
+            var generators = new Func<TransformManyBlock<int, int>>[]
+            {
+                () => new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable),
+                () => new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, new ExecutionDataflowBlockOptions { BoundedCapacity = 10 }),
+                () => new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, new ExecutionDataflowBlockOptions { BoundedCapacity = 10, MaxMessagesPerTask = 1 }),
+            };
+            foreach (var generator in generators)
+            {
+                DataflowTestHelpers.TestOfferMessage_ArgumentValidation(generator());
+
+                var target = generator();
+                DataflowTestHelpers.TestOfferMessage_AcceptsDataDirectly(target);
+                DataflowTestHelpers.TestOfferMessage_CompleteAndOffer(target);
+
+                target = generator();
+                await DataflowTestHelpers.TestOfferMessage_AcceptsViaLinking(target);
+                DataflowTestHelpers.TestOfferMessage_CompleteAndOffer(target);
+            }
+        }
+
+        [Fact]
+        public void TestPostAsyncEnumerable()
+        {
+            foreach (bool bounded in DataflowTestHelpers.BooleanValues)
+            {
+                var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, new ExecutionDataflowBlockOptions { BoundedCapacity = bounded ? 1 : -1 });
+                Assert.True(tb.Post(0));
+                tb.Complete();
+                Assert.False(tb.Post(0));
+            }
+        }
+
+        [Fact]
+        public Task TestCompletionTaskAsyncEnumerable()
+        {
+            return DataflowTestHelpers.TestCompletionTask(() => new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable));
+        }
+
+        [Fact]
+        public async Task TestLinkToOptionsAsyncEnumerable()
+        {
+            const int Messages = 1;
```

Comment:
```suggestion
            const int messages = 1;
```
private constants should be written in lower case

Reply:
> private constants should be written in lower case

That is not the style we use in the repo.  Constants of any kind are PascalCased.